floodlight/filter/filter.py

```diff
@@ -0,0 +1,223 @@
+from typing import Dict
+from floodlight.utils.types import Numeric
+
+import numpy as np
+import scipy.signal
+
+from floodlight import XY
+
+
+def _get_sequences_without_nans(data_column: np.ndarray) -> np.ndarray:
+    """Gets indices of sequences without NaNs from a sequence of data.
+
+    Parameters
+    ----------
+    data_column: np.ndarray
+        One-dimensional array of data with alternating sequences of Numbers and NaNs
+
+    Returns
+    -------
+    sequences: np.ndarray
+        Two-dimensional array with nx2 entrys of n sequences without NaNs and n[0] being
+        the first frame of the sequences and n[1] being the last frame (excluded) of the
+        sequence.
+    """
+
+    # check if even or odd sequences contain NaNs
```

Comment:
Even/odd in regard to them alternating. If `data[0]` is NaN, then `sequences[::2]` contain NaNs. These are the "even" sequences because they have even indices. If `data[0]` is not NaN, then the odd indexed `sequences[1::2]` contain NaNs. But I might remove this with regard to your point about making it more explicit.